modules/apps/layout/layout-item-selector-web/src/main/java/com/liferay/layout/item/selector/web/internal/display/context/LayoutItemSelectorViewDisplayContext.java

```diff
@@ -54,12 +90,70 @@ public boolean isPrivateLayout() {
 		return _privateLayout;
 	}
 
+	public boolean isShowBreadcrumb() {
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)_httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		PortletDisplay portletDisplay = themeDisplay.getPortletDisplay();
+
+		boolean showBreadcrumb = ParamUtil.getBoolean(
+			_httpServletRequest, "showBreadcrumb");
+
+		if (Objects.equals(
+				ItemSelectorPortletKeys.ITEM_SELECTOR,
+				portletDisplay.getPortletName()) ||
+			showBreadcrumb) {
+
+			return true;
+		}
+
+		return false;
+	}
+
 	public boolean isShowHiddenPages() {
 		return _layoutItemSelectorCriterion.isShowHiddenPages();
 	}
 
+	private BreadcrumbEntry _getHomeBreadcrumb() throws PortalException {
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)_httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		BreadcrumbEntry breadcrumbEntry = new BreadcrumbEntry();
+
+		breadcrumbEntry.setTitle(themeDisplay.getSiteGroupName());
+
+		return breadcrumbEntry;
+	}
+
+	private BreadcrumbEntry _getSitesAndLibrariesBreadcrumb()
```

Comment:
Same here (_getSitesAndLibrariesBreadcrumbEntry() )

Reply:
also, either include the full "Asset Libraries" name in the method (_getSitesAnd*Asset*LibrariesBreadcrumbEntry()), or use the alternative "Depot" (_getSitesAndDepotsBreadcrumbEntry()).

Also, if it returns a single breadcrumb entry, shouldn't the name be singular?

Reply:
hi @adolfopa I have renamed the methods. The plural in the method name stands for how that single entry is named: "Sites and Libraries" [is displayed](https://github.com/liferay-lima/liferay-portal/blob/f57668f6561cd267ed2a0fd2c8ccaec6122e30ee/modules/apps/layout/layout-item-selector-web/src/main/java/com/liferay/layout/item/selector/web/internal/display/context/LayoutItemSelectorViewDisplayContext.java#L137) on the screen.

Reply:
Makes sense. Thanks!